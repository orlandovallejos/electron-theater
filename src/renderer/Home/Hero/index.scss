.wrapper {
  background: pink;
  background-position: center; /* Center the image */
  background-repeat: no-repeat; /* Do not repeat the image */
  background-size: cover;
  min-height: 60vh;
  position: relative;

  .info {
    position: absolute;
    left: 50px;
    bottom: 100px;
    display: flex;
    flex-direction: column;
    min-width: 350px;

    .title {
      color: white;
      font-size: 60px;
      font-weight: 600;
      margin: 0 0 20px 0;
    }

    .btnWrapper {
      display: flex;
      justify-content: space-between;
    }
  }
}
