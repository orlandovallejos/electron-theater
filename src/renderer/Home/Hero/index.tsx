/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
import { MovieItem } from 'types';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
// Components
import Button from '../../common/Button';
// Styles
import HeroWrapper from './style';
// Helpers
import imageHelper from '../../../helpers/image';

type Props = {
  movie: MovieItem;
};
const Hero = (props: Props) => {
  const { movie } = props;

  return (
    <HeroWrapper
      style={{
        backgroundImage: `url(${imageHelper.getBackdropImage(
          movie.backdropPath
        )})`,
      }}
    >
      <div className="info">
        {/* <div>
          <span>2021</span>
          <span>Animation, Comedy, Family, Fantasy</span>
          <span>1h 35m</span>
        </div> */}
        <span className="title">{movie.title}</span>
        <div className="btn-wrapper">
          <Button types="primary" className="btn-trailer">
            Trailer
          </Button>
          <Button types="secondary">Watchlist</Button>
        </div>
      </div>
    </HeroWrapper>
  );
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default Hero;
