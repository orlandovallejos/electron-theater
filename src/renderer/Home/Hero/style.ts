/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import styled from 'styled-components';

const HeroWrapper = styled.div`
  background: pink;
  background-position: center; /* Center the image */
  background-repeat: no-repeat; /* Do not repeat the image */
  background-size: cover;
  min-height: 60vh;
  position: relative;

  &::after {
    content: '';
    height: 60vh;
    width: 100%;
    background: linear-gradient(0deg, #101010 0%, rgba(255, 255, 255, 0) 35%);
    display: table;
  }

  .info {
    position: absolute;
    left: 50px;
    bottom: 100px;
    display: flex;
    flex-direction: column;
    min-width: 350px;

    .title {
      color: white;
      font-size: 60px;
      font-weight: 600;
      margin: 0 0 20px 0;
    }

    .btn-wrapper {
      display: flex;
      justify-content: space-between;
    }
  }
`;
HeroWrapper.displayName = 'HeroWrapper';

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default HeroWrapper;
