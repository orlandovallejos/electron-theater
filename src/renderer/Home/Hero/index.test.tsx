/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
import shallow from 'enzyme/shallow';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import Hero from './index';

describe('<Hero />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Hero />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
