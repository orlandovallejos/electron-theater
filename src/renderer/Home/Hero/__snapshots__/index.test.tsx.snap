// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render correctly 1`] = `
"<HeroWrapper style={{...}}>
  <div className=\\"info\\">
    <span className=\\"title\\">
      My movie
    </span>
    <div className=\\"btn-wrapper\\">
      <Button types=\\"primary\\" className=\\"btn-trailer\\">
        Trailer
      </Button>
      <Button types=\\"secondary\\">
        Watchlist
      </Button>
    </div>
  </div>
</HeroWrapper>"
`;
